app/src/integrationTest/kotlin/maru/app/MaruFollowerDelayedStartTest.kt

```diff
@@ -0,0 +1,112 @@
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import testutils.Checks.assertMinedBlocks
+import testutils.PeeringNodeNetworkStack
+import testutils.besu.BesuFactory
+import testutils.besu.BesuTransactionsHelper
+import testutils.maru.MaruFactory
+
+class MaruFollowerDelayedStartTest {
+  private lateinit var cluster: Cluster
+  private lateinit var transactionsHelper: BesuTransactionsHelper
+  private val log = LogManager.getLogger(this.javaClass)
+  private val maruFactory = MaruFactory()
+
+  @BeforeEach
+  fun setUp() {
+    transactionsHelper = BesuTransactionsHelper()
+    cluster =
+      Cluster(
+        ClusterConfigurationBuilder().build(),
+        NetConditions(NetTransactions()),
+        ThreadBesuNodeRunner(),
+      )
+  }
+
+  @AfterEach
+  fun tearDown() {
+    cluster.close()
+  }
+
+  @Test
+  fun `follower can sync and accept transactions after delayed start`() {
+    val validatorStack = PeeringNodeNetworkStack()
+    PeeringNodeNetworkStack.startBesuNodes(cluster, validatorStack)
+
+    val validatorMaruApp =
+      maruFactory.buildTestMaruValidatorWithP2pPeering(
+        ethereumJsonRpcUrl = validatorStack.besuNode.jsonRpcBaseUrl().get(),
+        engineApiRpc = validatorStack.besuNode.engineRpcUrl().get(),
+        dataDir = validatorStack.tmpDir,
+      )
+    validatorStack.setMaruApp(validatorMaruApp)
+    validatorStack.maruApp.start()
+
+    val blocksToProduce = 5
+    repeat(blocksToProduce) {
+      transactionsHelper.run {
+        validatorStack.besuNode.sendTransactionAndAssertExecution(
+          logger = log,
+          recipient = createAccount("another account"),
+          amount = Amount.ether(100),
+        )
+      }
+    }
+
+    validatorStack.besuNode.assertMinedBlocks(blocksToProduce)
+
+    val followerStack =
+      PeeringNodeNetworkStack(
+        besuBuilder = { BesuFactory.buildTestBesu(validator = false) },
+      )
+    cluster.addNode(followerStack.besuNode)
+
+    val followerMaruApp =
+      maruFactory.buildTestMaruFollowerWithP2pPeering(
+        ethereumJsonRpcUrl = followerStack.besuNode.jsonRpcBaseUrl().get(),
+        engineApiRpc = followerStack.besuNode.engineRpcUrl().get(),
+        dataDir = followerStack.tmpDir,
+        validatorPortForStaticPeering = validatorStack.p2pPort,
+      )
+    followerStack.setMaruApp(followerMaruApp)
+    followerStack.maruApp.start()
+
+    followerStack.besuNode.assertMinedBlocks(blocksToProduce)
+
+    repeat(blocksToProduce) {
+      transactionsHelper.run {
+        followerStack.besuNode.sendTransactionAndAssertExecution(
+          logger = log,
+          recipient = createAccount("another account"),
+          amount = Amount.ether(100),
+        )
+      }
+    }
+
+    validatorStack.besuNode.assertMinedBlocks(2 * blocksToProduce)
+    followerStack.besuNode.assertMinedBlocks(2 * blocksToProduce)
+
+    followerStack.maruApp.stop()
```

Comment:
It's better to move them into a `@AfterEach`, because in case this test fails, the nodes may be left hanging